filters/builtin/headerfilter.go

```diff
@@ -244,15 +252,43 @@ func valueFromContext(
 	}
 }
 
+func renderValueTemplate(template *eskip.Template, ctx filters.FilterContext) (string, bool) {
```

Comment:
This function could be later refactored into context member method so that unified resolution algorithm can be reused by all filters. At the moment I've decided not to change context's contract.

Proposed approach allows multiple variable sources and custom logic, so e.g. it is easy to make request/response headers readily available for template resolution (prefix is not necessary):
```diff
diff --git a/filters/builtin/header_test.go b/filters/builtin/header_test.go
index 0387221..28687c4 100644
--- a/filters/builtin/header_test.go
+++ b/filters/builtin/header_test.go
@@ -147,6 +147,12 @@ func TestHeader(t *testing.T) {
                        args:    []interface{}{"X-Test-Name", "${one} ${two}"},
                        context: map[string]interface{}{"one": "1"},
                        valid:   true,
+               }, {
+                       msg:            "set request header from header (copy header)",
+                       args:           []interface{}{"X-Test-Bar", "${request:headers:X-Test-Foo}"},
+                       valid:          true,
+                       requestHeader:  http.Header{"X-Test-Foo": []string{"foo"}},
+                       expectedHeader: http.Header{"X-Test-Request-Foo": []string{"foo"}, "X-Test-Request-Bar": []string{"foo"}},
                }},
                "appendRequestHeader": {{
                        msg:            "append request header when none",
diff --git a/filters/builtin/headerfilter.go b/filters/builtin/headerfilter.go
index e655e0b..d7c2fcd 100644
--- a/filters/builtin/headerfilter.go
+++ b/filters/builtin/headerfilter.go
@@ -263,6 +263,12 @@ func renderValueTemplate(template *eskip.Template, ctx filters.FilterContext) (s
                                return s
                        }
                }
+               if strings.HasPrefix(key, "request:headers:") {
+                       h := key[len("request:headers:"):]
+                       if v := ctx.Request().Header.Get(h); v != "" {
+                               return v
+                       }
+               }
                missing = true
                return ""
        })
```